tests/test_dynamodb/test_dynamodb.py

```diff
@@ -5807,3 +5809,51 @@ def test_invalid_projection_expressions():
         ClientError, match="ProjectionExpression: Attribute name contains white space"
     ):
         client.scan(TableName=table_name, ProjectionExpression="not_a_keyword, na me")
+
+
+# https://github.com/getmoto/moto/pull/7073
+def test_dynamo_type_assertion_exception_not_raised_when_reloading_modules():
+    importlib.reload(sys.modules["moto"])
+    to_reload = [module for module in sys.modules if module.startswith("moto.")]
+    for module in to_reload:
+        importlib.reload(sys.modules[module])
+
+    importlib.reload(sys.modules["moto"])
+    to_reload = [module for module in sys.modules if module.startswith("moto.")]
+    for module in to_reload:
+        importlib.reload(sys.modules[module])
+
+    mock_dynamo = mock_dynamodb()
+    mock_dynamo.start()
+    dynamo_client = boto3.client("dynamodb")
```

Comment:
Our CI does not have a region configured, so we have to specify it here explicitly